src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ChannelBaseTests.4.0.0.cs

```diff
@@ -0,0 +1,258 @@
+            serviceProxy = client.Proxy;
+
+            // *** EXECUTE *** \\
+            // Note: public contract supports only async use on ChannelBase<T>
+            IAsyncResult ar = serviceProxy.BeginEcho(echoText, callback: null, asyncState: null);
+            string result = serviceProxy.EndEcho(ar);
+
+            // *** VALIDATE *** \\
+            Assert.True(String.Equals(echoText, result),
+                        String.Format("Expected response was '{0}' but actual was '{1}'", echoText, result));
+
+            // *** CLEANUP *** \\
+            ((ICommunicationObject)client).Close();
+            ((ICommunicationObject)serviceProxy).Close();
+        }
+        finally
+        {
+            // *** ENSURE CLEANUP *** \\
+            ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, (ICommunicationObject)client);
+        }
+    }
+
+    [WcfFact]
+    [OuterLoop]
+    public static void ChannelBaseOfT_Sync_Open_Close_Events_Fire()
+    {
+        MyClientBaseWithChannelBase client = null;
+        List<string> eventsCalled = new List<string>(4);
+        List<string> proxyEventsCalled = new List<string>(4);
+        IWcfServiceBeginEndGenerated serviceProxy = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            CustomBinding customBinding = new CustomBinding();
+            customBinding.Elements.Add(new TextMessageEncodingBindingElement());
+            customBinding.Elements.Add(new HttpTransportBindingElement());
+
+            client = new MyClientBaseWithChannelBase(customBinding, new EndpointAddress(Endpoints.HttpSoap12_Address));
+
+            // Listen to all events on the ClientBase and on the generated proxy
+            ClientBaseTestHelpers.RegisterForEvents(client, eventsCalled);
+
+            serviceProxy = client.Proxy;
+            ClientBaseTestHelpers.RegisterForEvents((ICommunicationObject)serviceProxy, proxyEventsCalled);
+
+            // *** EXECUTE *** \\
+            IAsyncResult ar = serviceProxy.BeginEcho("Hello", callback: null, asyncState: null);
+            string result = serviceProxy.EndEcho(ar);
+            ((ICommunicationObject)client).Close();
+
+            // *** VALIDATE *** \\
+
+            // We expect both the ClientBase and the generated proxy to have fired all the open/close events
+            string expected = "Opening,Opened,Closing,Closed";
+            string actual = String.Join(",", eventsCalled);
+
+            Assert.True(String.Equals(expected, actual),
+                        String.Format("Expected client to receive events '{0}' but actual was '{1}'", expected, actual));
+
+            actual = String.Join(",", proxyEventsCalled);
+            Assert.True(String.Equals(expected, actual),
+                        String.Format("Expected proxy to receive events '{0}' but actual was '{1}'", expected, actual));
+
+            // *** CLEANUP *** \\
```

Comment:
Maybe add a simple comment explaining that the closing of both proxies is supposed to have already happened and been verified above.